x/concentrated-liquidity/lp.go

```diff
@@ -134,6 +137,28 @@ func (k Keeper) withdrawPosition(ctx sdk.Context, owner sdk.AccAddress, position
 		return sdk.Int{}, sdk.Int{}, err
 	}
 
+	// If underlying lock exists in state, validate unlocked conditions are met before withdrawing liquidity.
+	// If unlocked conditions are met, remove the link between the position and the underlying lock.
+	if k.positionHasUnderlyingLockInState(ctx, position.PositionId) {
+		lockId, err := k.GetPositionIdToLock(ctx, positionId)
+		if err != nil {
+			return sdk.Int{}, sdk.Int{}, err
+		}
+
+		// Check if the underlying lock is mature.
+		lockIsMature, err := k.isLockMature(ctx, lockId)
+		if err != nil {
+			return sdk.Int{}, sdk.Int{}, err
+		}
+		if lockIsMature {
+			// Remove the link between the position and the underlying lock since the lock is mature.
+			k.RemovePositionIdToLock(ctx, positionId)
+		} else {
+			// Lock is not mature, return error.
+			return sdk.Int{}, sdk.Int{}, types.LockNotMatureError{LockId: lockId}
+		}
+	}
+
```

Comment:
Would like to see test cases added for this as well

Reply:
I am going to implement this into the next PR right now and link it here (in order to properly implement this test I want to have CreatePositionUnlocking and Locked methods)

Reply:
Not going to block this PR but in the future, please aim to add tests directly in the PR so that the reviewer can run them and make sure that edge cases are covered sufficiently

Reply:
I could not implement this test without the Unlocking and Locked methods that were added in the next PR. If I added them here, the next PR would have had nothing to review and would increased the scope of this PR